server/datastore/mysql/apple_mdm_test.go

```diff
@@ -14,6 +15,179 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestNewMDMAppleConfigProfile(t *testing.T) {
+	ds := CreateMySQLDS(t)
+
+	cases := []struct {
+		name string
+		fn   func(t *testing.T, ds *Datastore)
+	}{
+		{"TestNewMDMAppleConfigProfileDuplicateName", testNewMDMAppleConfigProfileDuplicateName},
+		{"TestNewMDMAppleConfigProfileDuplicateIdentifier", testNewMDMAppleConfigProfileDuplicateIdentifier},
+		{"TestDeleteMDMAppleConfigProfile", testDeleteMDMAppleConfigProfile},
+		{"TestListMDMAppleConfigProfiles", testListMDMAppleConfigProfiles},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			defer TruncateTables(t, ds)
+
+			c.fn(t, ds)
+		})
+	}
+}
+
+func testNewMDMAppleConfigProfileDuplicateName(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+	initialCP := storeDummyConfigProfileForTest(t, ds)
+
+	// cannot create another profile with the same name if it is on the same team
+	duplicateCP := fleet.MDMAppleConfigProfile{
+		Name:         initialCP.Name,
+		Identifier:   "DifferentIdentifierDoesNotMatter",
+		TeamID:       initialCP.TeamID,
+		Mobileconfig: initialCP.Mobileconfig,
+	}
+	_, err := ds.NewMDMAppleConfigProfile(ctx, duplicateCP)
+	expectedErr := &existsError{ResourceType: "MDMAppleConfigProfile.PayloadDisplayName", Identifier: initialCP.Name, TeamID: &initialCP.TeamID}
+	require.ErrorContains(t, err, expectedErr.Error())
+
+	// can create another profile with the same name if it is on a different team
+	duplicateCP.TeamID += 1
+	newCP, err := ds.NewMDMAppleConfigProfile(ctx, duplicateCP)
+	require.NoError(t, err)
+	checkConfigProfile(t, &duplicateCP, newCP)
+	storedCP, err := ds.GetMDMAppleConfigProfile(ctx, newCP.ProfileID)
+	require.NoError(t, err)
+	checkConfigProfile(t, newCP, storedCP)
+}
+
+func testNewMDMAppleConfigProfileDuplicateIdentifier(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+	initialCP := storeDummyConfigProfileForTest(t, ds)
+
+	// cannot create another profile with the same identifier if it is on the same team
+	duplicateCP := fleet.MDMAppleConfigProfile{
+		Name:         "DifferentNameDoesNotMatter",
+		Identifier:   initialCP.Identifier,
+		TeamID:       initialCP.TeamID,
+		Mobileconfig: initialCP.Mobileconfig,
+	}
+	_, err := ds.NewMDMAppleConfigProfile(ctx, duplicateCP)
+	expectedErr := &existsError{ResourceType: "MDMAppleConfigProfile.PayloadIdentifier", Identifier: initialCP.Identifier, TeamID: &initialCP.TeamID}
+	require.ErrorContains(t, err, expectedErr.Error())
```

Comment:
Nice, I always forget that this `ErrorContains` exists and I end up using `Contains(err.Error()...` :)